wallet/wallet.go

```diff
@@ -4484,19 +4484,25 @@ func (w *Wallet) AbandonTransaction(ctx context.Context, hash *chainhash.Hash) e
 
 // PublishTransaction saves (if relevant) and sends the transaction to the
 // consensus RPC server so it can be propagated to other nodes and eventually
-// mined.  If the send fails, the transaction is not added to the wallet.
-func (w *Wallet) PublishTransaction(ctx context.Context, tx *wire.MsgTx, n NetworkBackend) (*chainhash.Hash, error) {
+// mined.  If the transaction is not relevant to the wallet or the send fails,
+// the transaction is not added to the wallet.  If specified, `permitHighFees`
+// overrides `w.allowHighFees`.
+func (w *Wallet) PublishTransaction(ctx context.Context, tx *wire.MsgTx, permitHighFees *bool, n NetworkBackend) (*chainhash.Hash, error) {
```

Comment:
allowHighFees (use this name) should not be a pointer. it can be dereferenced in the rpc method handler (and no need to check for nil, it is filled in with a pointer to a default value if none is specified by dcrjson).

Reply:
There is a `w.allowHighFees` property that I'll want to default to if the rpc request does not specify a value. A gRPC method also calls this method and currently passes nil to allow wallet use the user-configured value.

Reply:
I saw that, but we need to default to the RPC server's default (it's defined in the command's struct tags, appears in help, etc).  The wallet's default shouldn't come into play here.

Reply:
> The wallet's default shouldn't come into play here.

That makes sense.

So expose `w.allowHighFees` via a `w.AllowHighFees() bool` method to ensure it can only be set/modified from within the wallet package, its value is relied upon in a number of other sites especially via calls to `w.checkHighFees(...)`, I think it'd be better to limit its write scope.